asreviewcontrib/datatools/dedup.py

```diff
@@ -1,38 +1,44 @@
 import argparse
 
+import numpy as np
 import pandas as pd
 
-from asreview.data import load_data
 
+def dedup(asdata, pid='doi'):
+    initial_length = len(asdata.df)
 
-def dedup(input_path, output_path=None):
-    # read data in ASReview data object
-    asdata = load_data(input_path)
+    if pid in asdata.df.columns:
+        # replace NaN values with empty string to prevent astype(str) from creating literal 'NaN' string
+        asdata.df[pid].fillna('', inplace=True)
+
+        # convert to string to support non-string types, strip whitespaces and replace empty strings with NaN values
+        s_pid = asdata.df[pid].astype(str).str.strip().replace("", np.nan)
+
+        # remove records based on duplicate PIDs
+        asdata.df = asdata.df[(~s_pid.duplicated()) | (s_pid.isnull())].reset_index(drop=True)
+    else:
+        print(f"Not using {pid} for deduplication because there is no such data.")
 
     # get the texts and clean them
     s = pd.Series(asdata.texts) \
         .str.replace("[^A-Za-z0-9]", "", regex=True) \
         .str.lower()
 
-    # remove the records
-    asdata.df = asdata.df[~s.duplicated()]
+    # remove records based on duplicate texts
+    asdata.df = asdata.df[~s.duplicated()].reset_index(drop=True)
```

Comment:
Good one, do we need to reset the index?

Reply:
The dedup function returns an ASReview data object, so the dataframe it contains could be used further in some pipeline, I thought it would be neat to reset the indexes before returning the ASReview data object.

Reply:
Agreed.